app/main.py

```diff
@@ -1,36 +1,58 @@
-from fastapi import FastAPI, Depends, HTTPException, status
+import sys
+import os
+
+# Add the parent directory to sys.path
+sys.path.append(os.path.dirname(os.path.dirname(__file__)))
+
+from fastapi import FastAPI, Depends, HTTPException, status, Security
+from fastapi.security import OAuth2PasswordRequestForm
+from pydantic import EmailStr
+from typing import List, Annotated
 from sqlalchemy.orm import Session
-from app.database import SessionLocal, engine
-from app.models import Base, UserDB
-from app.schemas import UserCreate, UserOut
-from app.crud import create_user, get_user_by_email
+from database import engine, get_db, Base
+from models.user import UserDB
+from schemas.user import UserCreate, UserOut, User
+from schemas.task import TaskCreate, TaskOut, TaskStatus
+from schemas.token import Token
+from crud.user import create_user, get_user_by_email
+from crud.task import create_task, get_tasks, get_task, update_task, delete_task, get_tasks_by_priority, get_tasks_by_status, get_task_by_deadline
+from app.utils.auth import get_current_user, create_access_token, verify_password
+from datetime import datetime
 
-# Create all tables in the database
 Base.metadata.create_all(bind=engine)
 
 app = FastAPI()
 
-# Dependency to get DB session
-def get_db():
-    db = SessionLocal()
-    try:
-        yield db
-    finally:
-        db.close()
 
-@app.post("/users/", response_model=UserOut, status_code=status.HTTP_201_CREATED)
+@app.post("/register/", response_model=UserOut, status_code=status.HTTP_201_CREATED)
 def register_user(user: UserCreate, db: Session = Depends(get_db)):
-    db_user = get_user_by_email(db, user.e_mail)
-    if db_user:
+    if get_user_by_email(db, user.email_address):
         raise HTTPException(
-            status_code=status.HTTP_400_BAD_REQUEST,
-            detail="Email already registered",
+            status_code=status.HTTP_409_CONFLICT,
+            detail="Email already in use",
         )
-    return create_user(db=db, user=user)
 
-@app.get("/users/{e_mail}", response_model=UserOut)
-def read_user(e_mail: str, db: Session = Depends(get_db)):
-    user = get_user_by_email(db, e_mail)
+    return create_user(db, user)
+
+@app.post("/token", response_model=Token)
+def login_for_access_token(
+    form_data: Annotated[OAuth2PasswordRequestForm, Depends()], db: Session = Depends(get_db)
+):
+
+    user = get_user_by_email(db, form_data.username)
+
+    if not user or not verify_password(form_data.password, user.password):
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Incorrect username or password",
+            headers={"Authorization": "Bearer"},
+        )
+    access_token = create_access_token(data={"sub": user.email_address})
+    return {"access_token": access_token, "token_type": "Bearer"}
+
+@app.get("/users/{email_address}", response_model=UserOut)
```

Comment:
maaaaybe use user_id to fetch user, just a suggestion, cuz you have an index on ids :)